src/multirust-cli/self_update.rs

```diff
@@ -0,0 +1,1000 @@
+            let rcfiles_str = rcfiles.join("\n");
+            Ok(format!(pre_install_msg_unix!(),
+                       cargo_home_bin = cargo_home.display(),
+                       rcfiles = rcfiles_str))
+        } else {
+            Ok(format!(pre_install_msg_unix_norcfiles!(),
+                       cargo_home_bin = cargo_home.display()))
+        }
+    } else {
+        Ok(format!(pre_install_msg_win!(),
+                   cargo_home_bin = cargo_home.display()))
+    }
+}
+
+// Before multirust-rs installed bins to $CARGO_HOME/bin it installed
+// them to $MULTIRUST_HOME/bin. If those bins continue to exist after
+// upgrade and are on the $PATH, it would cause major confusion. This
+// method silently deletes them.
+fn cleanup_legacy() -> Result<()> {
+    let legacy_bin_dir = try!(legacy_multirust_home_dir()).join("bin");
+
+    for tool in TOOLS.iter().cloned().chain(Some("multirust")) {
+        let ref file = legacy_bin_dir.join(&format!("{}{}", tool, EXE_SUFFIX));
+        if file.exists() {
+            try!(utils::remove_file("legacy-bin", file));
+        }
+    }
+
+    return Ok(());
+
+    #[cfg(unix)]
+    fn legacy_multirust_home_dir() -> Result<PathBuf> {
+        multirust_dir()
+    }
+
+    #[cfg(windows)]
+    fn legacy_multirust_home_dir() -> Result<PathBuf> {
+        use multirust_utils::raw::windows::{
+            get_special_folder, FOLDERID_Profile
+        };
+
+        Ok(get_special_folder(&FOLDERID_Profile).unwrap_or(PathBuf::from(".")))
+    }
+}
+
+fn install_bins() -> Result<()> {
+    let ref bin_path = try!(get_cargo_home()).join("bin");
+    let ref this_exe_path = try!(utils::current_exe());
+    let ref multirust_path = bin_path.join(&format!("multirust{}", EXE_SUFFIX));
+
+    try!(utils::ensure_dir_exists("bin", bin_path, ntfy!(&NotifyHandler::none())));
+    // NB: Even on Linux we can't just copy the new binary over the (running)
+    // old binary; we must unlink it first.
+    if multirust_path.exists() {
+        try!(utils::remove_file("multirust-bin", multirust_path));
+    }
+    try!(utils::copy_file(this_exe_path, multirust_path));
+    try!(utils::make_executable(multirust_path));
+
+    // Hardlink all the Rust exes to the multirust exe. Using hardlinks
+    // because they work on Windows.
+    for tool in TOOLS {
+        let ref tool_path = bin_path.join(&format!("{}{}", tool, EXE_SUFFIX));
+        try!(utils::hardlink_file(multirust_path, tool_path))
+    }
+
+    Ok(())
+}
+
+fn maybe_install_rust_stable(verbose: bool) -> Result<()> {
+    let ref cfg = try!(common::set_globals(verbose));
+
+    // If this is a fresh install (there is no default yet)
+    // then install stable and make it the default.
+    if try!(cfg.find_default()).is_none() {
+        let stable = try!(cfg.get_toolchain("stable", false));
+        try!(stable.install_from_dist());
+        try!(cfg.set_default("stable"));
+    } else {
+        info!("updating existing installation");
```

Comment:
It's just indicating whether setup is in 'install mode' or 'upgrade mode'. The `installation` here is not referring to rustc but to rustup. If there weren't a bunch of other status messages printed by the other branch it might contain the line `info!("doing a fresh install")`.